src/framework/components/audiosource/audiosource_component.js

```diff
@@ -230,6 +230,21 @@ pc.extend(pc, function () {
 
         onEnable: function () {
             AudioSourceComponent._super.onEnable.call(this);
+
+            // load assets that haven't been loaded yet
+            var assets = this.data.assets;
+            if (assets) {
+                for (var i = 0, len = assets.length; i < len; i++) {
+                    var asset = assets[i];
+                    if (! (asset instanceof pc.Asset))
+                        asset = this.system.app.assets.get(asset);

```

Comment:
just for readability like in animation_component, store assets registry in `var` above.